rpc_util.go

```diff
@@ -148,7 +148,9 @@ func encode(msg proto.Message, pf payloadFormat) ([]byte, error) {
 		}
 		length = uint32(len(b))
 	}
-	if err := binary.Write(&buf, binary.BigEndian, length); err != nil {
+	var szHdr [4]byte
+	binary.BigEndian.PutUint32(szHdr[:], length)
+	if _, err := buf.Write(szHdr[:]); err != nil {
```

Comment:
The only difference I can see is that i) there is a [8]byte alloc in the original code; [4]byte in your code (there is no additional alloc for slice bs because it just uses b's memory); ii) the original code has two type switches, your code does not have. Your code is the final solution but is okay to check in.